src/reconstruction/community.jl

```diff
@@ -0,0 +1,388 @@
+
+    # adds to the original community data, could possibly reset?
+    I, V = findnz(balance(community))
+    b = sparsevec(I, V, n_metabolites_total)
+    I, V = findnz(objective(community))
+    c = sparsevec(I, V, n_reactions_total)
+
+    return CoreModel(S, b, c, lbs, ubs, rxns, mets)
+end
+
+"""
+    join(models::Vector{M}, 
+        exchange_rxn_ids::Vector{String}, 
+        exchange_met_ids::Vector{String}; 
+        add_biomass_objective=false, 
+        biomass_ids::Vector{String}, 
+        species_names=String[]
+    ) 
+
+Return a `CoreModel` representing the community model of `models` joined through
+their `exchange_rxn_ids` and `exchange_met_ids`. These exchange reactions and
+metabolites link to environmental metabolites and reactions.
+Optionally specify `species_names` to append a specific name to each reaction
+and metabolite of an organism for easier reference (default is `species_i` for
+each model index i in `models`). Note, the bounds of the environmental variables
+are all set to zero. Thus, to run a simulation you need to constrain them
+appropriately. All the other bounds are inherited from the models used to
+construct the community model.
+
+If `add_biomass_objective` is true then `biomass_ids` needs to be supplied as
+well. This creates a model with an extra reaction added to the end of the
+stoichiometric matrix (last column) that can be assigned as the objective
+reaction. It also creates biomass "metabolites" that can be used in this
+objective reaction. Note, this reaction is unspecified, further action needs to
+be taken to specify it, e.g. assign weights to the last column of the
+stoichiometric matrix in the rows corresponding to the biomass metabolites.
+
+To further clarify how this `join` works. Suppose you have 2 organisms with
+stoichiometric matrices S₁ and S₂ and you want to link them with
+`exchange_met_ids = [em₁, em₂, em₃, ...]` and `exchange_rxn_ids = [er₁, er₂,
+er₃, ...]`. Then a new community stoichiometric matrix is constructed that looks
+like this:
+```
+            _      er₁  er₂  er₃  ...  b_
+           |S₁                           |
+           |   S₂                        |
+        em₁|                             |
+S   =   em₂|                             |
+        em₃|                             |
+        ...|                             |
+        bm₁|                             |  
+        bm₂|_                           _|
+
+```
+The exchange reactions in each model get linked to environmental metabolites,
+`emᵢ`, and these get linked to environmental exchanges, `erᵢ`. These `erᵢ`
+behave like normal single organism exchange reactions. When
+`add_biomass_objective` is true each model's biomass becomes a pseudo-metabolite
+(`bmᵢ`). These can be weighted in column `b`, called the `community_biomass`
+reaction in the community model, if desired. Refer to the tutorial if this is
+unclear.
+
+# Example
+```
+model_path = joinpath("..","models","e_coli_core.json")
```

Comment:
I still believe that we should have doctests ...

Reply:
easily done! How do I load "e_coli_core.json" for the doctests?

Reply:
in the doctests.jl file

Reply:
done